lib/YaST/workarounds.pm

```diff
@@ -0,0 +1,27 @@
+# Copyright 2015-2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package YaST::workarounds;
+
+
+use strict;
+use warnings;
+use Exporter;
+use Mojo::Base 'Exporter', -signatures;
+use testapi;
+use version_utils;
+
+
+our @EXPORT = qw(workaround_screen_refresh);
+
+sub workaround_screen_refresh {
+    my ($needle) = shift;
+    if (!check_screen($needle, 180) && (is_sle('>=15-SP4'))) {
+        record_soft_failure('bsc#1204176 - Resizing window as workaround for YaST content not loading');
+        send_key('shift-f3', wait_screen_change => 1);
+        send_key('esc');
+        send_key_until_needlematch($needle, 'esc', 10, 2);
```

Comment:
with a wait_screen_change should do, it is better to avoid send the same key so many times, we might find ourselves in a complete different screen...

Reply:
The extra `send_key('esc')` is there for a reason: I figured out that "send_key_until_needlematch" does not send a key at all if the needle matches already. So its more a "while_not_needlematch_send_key" function than what the name suggests. Nevertheless, I will try it out. The VRs still show that sometimes the workaround is applied, but does not work.

Reply:
I think you don't need that command, just  wrap it as usual: `wait_screen_change { send_key 'esc'};`